src/app/state/persist/index.ts

```diff
@@ -0,0 +1,112 @@
+import { PayloadAction } from '@reduxjs/toolkit'
+import { createSlice } from 'utils/@reduxjs/toolkit'
+import {
+  PersistedRootState,
+  PersistState,
+  SetUnlockedRootStatePayload,
+  StringifiedKeyWithSalt,
+} from './types'
+import { RootState } from 'types'
+import { runtimeIs } from 'config'
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { createPersistedRootReducer } from 'store/reducers'
+
+export const STORAGE_FIELD = 'oasis_wallet_persist_v1'
+
+/** Syncing tabs is only needed in web app, not in extension. */
+export const needsSyncingTabs = runtimeIs === 'webapp'
+
+// Simulate with `delete window.BroadcastChannel`
+export const isSyncingTabsSupported = typeof BroadcastChannel === 'function'
+
+export function getInitialState(): PersistState {
+  return {
+    hasPersistedProfiles: !!localStorage.getItem(STORAGE_FIELD),
```

Comment:
security review: only uses of `localStorage` in master branch are related to theme selection

non-security: tests use `window.localStorage`, app code uses `localStorage`, previous app code uses `window.localStorage && localStorage`